src/core_write.cpp

```diff
@@ -207,6 +207,14 @@ void TxToUniv(const CTransaction& tx, const uint256& block_hash, UniValue& entry
                 txinwitness.push_back(HexStr(item));
             }
             in.pushKV("txinwitness", std::move(txinwitness));
+            CPubKey pubkey(tx.vin[i].scriptWitness.stack.back());
+            if (!pubkey.IsFullyValid()){
```

Comment:
This isn't an indicator of p2wsh... Maybe use d1f0ff0e1fed2c4124d0d81f18daf7b76a770dda or similar?

Reply:
code rewritten based on this comment